src/serialization/sb2.js

```diff
@@ -245,17 +245,42 @@ const globalBroadcastMsgStateGenerator = (function () {
  * @param {ImportedExtensionsInfo} extensions - (in/out) parsed extension information will be stored here.
  */
 const parseMonitorObject = (object, runtime, targets, extensions) => {
-    let target = null;
+    // In scratch 2.0, there are two monitors that now correspond to extension
+    // blocks (tempo and video motion/direction). In the case of the
+    // video motion/direction block, this reporter is not monitorable in Scratch 3.0.
+    // In the case of the tempo block, we should import it and load the music extension
+    // only when the monitor is actually visible.
+
     const opcode = specMap[object.cmd].opcode;
     const extIndex = opcode.indexOf('_');
     const extID = opcode.substring(0, extIndex);
 
-    // All non-core extensions should be added by blocks at this point
-    // We can assume this is an unintended monitor and skip parsing if it belongs to a non-core extension
-    if (CORE_EXTENSIONS.indexOf(extID) === -1) {
-        if (extID !== '') return;
+    if (extID === 'videoSensing') {
+        return;
+    } else if (CORE_EXTENSIONS.indexOf(extID) === -1 && extID !== '' &&
+        !extensions.extensionIDs.has(extID) && !object.visible) {
```

Comment:
This line sets up a hard ordering requirement between block loading/extension finding and monitor parsing. I don't know if there is something we can do to enforce this so that it doesn't get shuffled in the future, but it might be worth at least noting in the JSdoc at the top of parseMonitorObject that calling this before you've loaded the projects blocks will result in undefined behavior.

Reply:
Sounds good!